### DISCUSSION TITLE
Missing ISO/RTO information in out_eia__yearly_plants table

### DISCUSSION THREAD

**Discussion by nomineb** [catalyst-cooperative/pudl#3771] [Help Me!]
Hello,

I'm trying to associate plant IDs in CEMS dataset from year 2017 with their respective ISO/RTO and NERC codes by merging it with the information from `out_eia__yearly_plants` table. However, when I set the report date to 2017-01-01, the resulting query contains no information of iso_rto_code but 10,127 rows have nerc_region information (see screenshot attached).
<img width="752" alt="Screen Shot 2024-08-07 at 12 52 41 PM" src="https://github.com/user-attachments/assets/fd53fb74-2c80-4d63-bf67-1be96c570802">
<img width="775" alt="Screen Shot 2024-08-07 at 12 53 03 PM" src="https://github.com/user-attachments/assets/9e38a389-043e-4ca2-a954-d02ed9a480e2">

Is there another table that contains more complete ISO/RTO, NERC region names for plant IDs in CEMS data?

**Comment by zaneselvans** [comment]
For some reason the `iso_rto_code` column was only reported by EIA for 2010-2012, so we've retained it, but as those associations can change as the ISO/RTO territories shift over time, we didn't try to forward or backfill the data.

It's not exactly the same thing, but you might try using the `balancing_authority_code_eia` instead. The ISO/RTOs show up in there (alongside the other smaller BAs) and should cover much more of the dataset. For example (in duckdb):

```sql
SELECT
    plant_id_eia,
    plant_name_eia,
    report_date,
    balancing_authority_code_eia,
    iso_rto_code,
    nerc_region
FROM out_eia__yearly_plants
USING SAMPLE 20
ORDER BY report_date;
```
| plant_id_eia |         plant_name_eia         | report_date | balancing_authority_code_eia | iso_rto_code | nerc_region |
|-------------:|--------------------------------|-------------|------------------------------|--------------|-------------|
| 10382        | Lumberton                      | 2001-01-01  | CPLE                         |              | SERC        |
| 2103         | Labadie                        | 2003-01-01  | MISO                         |              | MAIN        |
| 4043         | Appleton                       | 2006-01-01  | MISO                         |              | RFC         |
| 2854         | Yankee Street                  | 2006-01-01  | PJM                          |              | RFC         |
| 7888         | Parkside                       | 2006-01-01  | MISO                         |              | RFC         |
| 50281        | San Gorgonio Farms Wind Energy | 2007-01-01  | CISO                         |              | WECC        |
| 7901         | Byron                          | 2008-01-01  | MISO                         |              | RFC         |
| 50024        | IBM San Jose Standby Generator | 2010-01-01  | CISO                         | CAISO        | WECC        |
| 56479        | Sand Bluff                     | 2012-01-01  | ERCO                         | ERCOT        | TRE         |
| 6055         | Big Cajun 2                    | 2013-01-01  | MISO                         |              | SERC        |
| 1620         | Deerfield 5                    | 2014-01-01  | ISNE                         |              | NPCC        |
| 57516        | DeWind Novus                   | 2014-01-01  | SWPP                         |              | SPP         |
| 59647        | Stillwater Energy Center       | 2016-01-01  | SWPP                         |              | SPP         |
| 54809        | University of Washington       | 2017-01-01  | SCL                          |              | WECC        |
| 58239        | Kent County-Kennedyville       | 2018-01-01  | PJM                          |              | RFC         |
| 712          | Lloyd Shoals                   | 2021-01-01  | SOCO                         |              | SERC        |
| 59049        | Whitcomb Solar Farm            | 2021-01-01  | ISNE                         |              | NPCC        |
| 6801         | Indian River                   | 2022-01-01  |                              |              | UNK         |
| 65024        | Novel Brock Solar LLC          | 2022-01-01  | MISO                         |              | MRO         |
| 2511         | EF Barrett                     | 2023-01-01  | NYIS                         |              | NPCC        |

Looking just at the records where there is a value for `iso_rto_code` it seems like there a close (though not perfect) correspondence between BA and ISO.  Some mismatches associated with SPP in particular...

| plant_id_eia |         plant_name_eia         | report_date | balancing_authority_code_eia | iso_rto_code | nerc_region |
|-------------:|--------------------------------|-------------|------------------------------|--------------|-------------|
| 10700        | Old Town Division              | 2010-01-01  | ISNE                         | ISONE        | NPCC        |
| 539          | Rocky River                    | 2010-01-01  | ISNE                         | ISONE        | NPCC        |
| 510          | Calpine Geysers-Sonoma         | 2010-01-01  | CISO                         | CAISO        | WECC        |
| 54678        | Sidney A Murray Jr Hydro       | 2010-01-01  | MISO                         | SPP          | SERC        |
| 2023         | Windom                         | 2010-01-01  | MISO                         | MISO         | MRO         |
| 55285        | Wolf Hill Energy               | 2010-01-01  | PJM                          | PJM          | RFC         |
| 323          | Bishop Creek 2                 | 2010-01-01  | CISO                         | CAISO        | WECC        |
| 10708        | Lost Creek I                   | 2010-01-01  | CISO                         | CAISO        | WECC        |
| 57555        | Pio Pico Energy Center         | 2010-01-01  | CISO                         | CAISO        | WECC        |
| 56572        | Bethlehem Renewable Energy LLC | 2010-01-01  | PJM                          | PJM          | RFC         |
| 2011         | Sleepy Eye                     | 2010-01-01  | MISO                         | MISO         | MRO         |
| 2039         | Elk River                      | 2011-01-01  | MISO                         | MISO         | MRO         |
| 2881         | FirstEnergy Stryker            | 2011-01-01  | PJM                          | PJM          | RFC         |
| 10826        | Winsor Dam Power Station       | 2011-01-01  | ISNE                         | ISONE        | NPCC        |
| 54757        | Grand Blanc                    | 2011-01-01  | MISO                         | MISO         | RFC         |
| 2013         | Spring Valley                  | 2011-01-01  | MISO                         | MISO         | MRO         |
| 6419         | Eufaula Dam                    | 2011-01-01  | SPA                          | SPP          | SPP         |
| 3631         | Sam Rayburn                    | 2011-01-01  | ERCO                         | ERCOT        | TRE         |
| 50885        | Gloucester                     | 2011-01-01  | PJM                          | PJM          | RFC         |
| 56031        | Fox Energy Center              | 2011-01-01  | MISO                         | MISO         | RFC         |
| 54569        | Bayville Central Facility      | 2011-01-01  | PJM                          | PJM          | RFC         |
| 2512         | East Hampton                   | 2011-01-01  | NYIS                         | NYISO        | NPCC        |
| 553          | Stevenson                      | 2011-01-01  | ISNE                         | ISONE        | NPCC        |
| 418          | Kelly Ridge                    | 2011-01-01  | CISO                         | CAISO        | WECC        |
| 55738        | Buchanan Generating Units 1& 2 | 2012-01-01  | PJM                          | PJM          | RFC         |
| 7847         | Coon Rapids II                 | 2012-01-01  | MISO                         | MISO         | MRO         |
| 1710         | J H Campbell                   | 2012-01-01  | MISO                         | MISO         | RFC         |
| 10045        | Kinsleys Landfill Inc          | 2012-01-01  | PJM                          | PJM          | RFC         |
| 58216        | J.R. Simplot Company           | 2012-01-01  | CISO                         | CAISO        | WECC        |
| 57958        | LanWest Solar Farm LLC         | 2012-01-01  | CISO                         | CAISO        | WECC        |
| 2573         | Heuvelton                      | 2012-01-01  | NYIS                         | NYISO        | NPCC        |
| 7497         | Croswell                       | 2012-01-01  | MISO                         | MISO         | RFC         |
| 50647        | Covanta Indianapolis Inc       | 2012-01-01  | MISO                         | MISO         | RFC         |

**Reply by nomineb** [reply]
Thank you! I'll try that.
